### PR TITLE
Fixups from #3324

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Fixes some bugs and other issues, including unnecessary metadata options and uneven info in `metadata.yaml` files, introduced in #3324

### CODE REVIEW

**Review by berndverst — APPROVED** [review]

**Comment by ItalyPaleAle** [comment]
FYI @sicoyle

**Comment by sicoyle** [comment]
reviewing rn. Thanks for the tag here :)

File: bindings/postgres/metadata.yaml [hunk 1]
```diff
@@ -47,11 +47,20 @@ builtinAuthenticationProfiles:
...
           This authentication method only works with AWS Relational Database Service for PostgreSQL databases.
+      - name: connectionString
+        required: true
+        sensitive: true
+        description: |
+          The connection string for the PostgreSQL database
+          This must contain the user, which corresponds to the name of the user created inside PostgreSQL that maps to the AWS IAM policy. This connection string should not contain any password. Note that the database name field is denoted by dbname with AWS.
+        example: |
+          "host=mydb.postgres.database.aws.com user=myapplication port=5432 dbname=dapr_test sslmode=require"
+        type: string
```
Comment (sicoyle) [inline bindings/postgres/metadata.yaml:58]:
```suggestion
          The connection string for the PostgreSQL database.
          This must contain the user, which corresponds to the name of the user created inside PostgreSQL that maps to the AWS IAM policy. This connection string should not contain any password. Note that the database name field is denoted by dbname with AWS.
        example: |
          "host=mydb.postgres.database.aws.com user=myapplication port=5432 dbname=dapr_test sslmode=require"
        type: string
```

File: common/authentication/aws/aws.go [hunk 2]
```diff
@@ -138,7 +136,7 @@ func (opts *AWSIAMAuthOptions) GetAccessToken(ctx context.Context) (string, erro
 	return authenticationToken, nil
 }
 
-func (opts *AWSIAMAuthOptions) InitiateAWSIAMAuth(ctx context.Context) error {
+func (opts *AWSIAMAuthOptions) InitiateAWSIAMAuth() error {
```
Comment (sicoyle) [inline common/authentication/aws/aws.go:139]: why are you removing ctx here?

File: configuration/postgres/metadata.yaml [hunk 3]
```diff
@@ -37,11 +37,20 @@ builtinAuthenticationProfiles:
...
           Must be set to `true` to enable the component to retrieve access tokens from AWS IAM.
           This authentication method only works with AWS Relational Database Service for PostgreSQL databases.
+      - name: connectionString
+        required: true
+        sensitive: true
+        description: |
+          The connection string for the PostgreSQL database
+          This must contain the user, which corresponds to the name of the user created inside PostgreSQL that maps to the AWS IAM policy. This connection string should not contain any password. Note that the database name field is denoted by dbname with AWS.
+        example: |
+          "host=mydb.postgres.database.aws.com user=myapplication port=5432 dbname=dapr_test sslmode=require"
```
Comment (sicoyle) [inline configuration/postgres/metadata.yaml:47]:
```suggestion
          The connection string for the PostgreSQL database.
          This must contain the user, which corresponds to the name of the user created inside PostgreSQL that maps to the AWS IAM policy. This connection string should not contain any password. Note that the database name field is denoted by dbname with AWS.
        example: |
          "host=mydb.postgres.database.aws.com user=myapplication port=5432 dbname=dapr_test sslmode=require"
```

**Review by sicoyle — APPROVED** [review]
few minor things. Thanks for the adjustments and cleanup :)

File: common/authentication/aws/aws.go (hunk 2 above)
Comment (ItalyPaleAle) [reply common/authentication/aws/aws.go:139]: It was unused

**Comment by marcduiker** [comment]
@holopin-bot @ItalyPaleAle Enjoy the Dapr components badge!

**Pull request merged**